src/App.tsx

```diff
@@ -1,16 +1,87 @@
 import './App.scss';
+import { useEffect, useState } from 'react';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
+import { IsMovie } from './types/Movie';
 import moviesFromServer from './api/movies.json';
 
 export const App = () => {
+  const [movies, setMovies] = useState(moviesFromServer);
+  const [isButtonDisabled, setIsButtonDisabled] = useState(false);
+  const [touchedMovies, setTouchedMovies] = useState<IsMovie>(
+    {
+      title: false,
+      description: false,
+      imgUrl: false,
+      imdbUrl: false,
+      imdbId: false,
+    },
+  );
+  const [newMovie, setNewMovie] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const addOn = () => {
+    setMovies([...movies, newMovie]);
+
+    if (newMovie.title.length === 0
```

Comment:
Use logical operators an destructuring for objects
```suggestion
    if (!title.length
```